tests/pytests/unit/states/test_keystore.py

```diff
@@ -0,0 +1,364 @@
+        "Issuer": {
+            "C": "JP",
+            "organizationName": "Frank4DD",
+            "CN": "Frank4DD Web CA",
+            "SP": "Tokyo",
+            "L": "Chuo-ku",
+            "emailAddress": "support@frank4dd.com",
+            "OU": "WebCert Support",
+        },
+        "Issuer Hash": "92:DA:45:6B",
+        "Not Before": "2012-08-22 05:26:54",
+        "Subject": {
+            "C": "JP",
+            "SP": "Tokyo",
+            "organizationName": "Frank4DD",
+            "CN": "www.example.com",
+        },
+    }
+
+    name = "keystore.jks"
+    passphrase = "changeit"
+    entries = [
+        {
+            "alias": "stringhost",
+            "certificate": """-----BEGIN CERTIFICATE-----
+               MIICEjCCAXsCAg36MA0GCSqGSIb3DQEBBQUAMIGbMQswCQYDVQQGEwJKUDEOMAwG
+               A1UECBMFVG9reW8xEDAOBgNVBAcTB0NodW8ta3UxETAPBgNVBAoTCEZyYW5rNERE
+               MRgwFgYDVQQLEw9XZWJDZXJ0IFN1cHBvcnQxGDAWBgNVBAMTD0ZyYW5rNEREIFdl
+               YiBDQTEjMCEGCSqGSIb3DQEJARYUc3VwcG9ydEBmcmFuazRkZC5jb20wHhcNMTIw
+               ODIyMDUyNjU0WhcNMTcwODIxMDUyNjU0WjBKMQswCQYDVQQGEwJKUDEOMAwGA1UE
+               CAwFVG9reW8xETAPBgNVBAoMCEZyYW5rNEREMRgwFgYDVQQDDA93d3cuZXhhbXBs
+               ZS5jb20wXDANBgkqhkiG9w0BAQEFAANLADBIAkEAm/xmkHmEQrurE/0re/jeFRLl
+               8ZPjBop7uLHhnia7lQG/5zDtZIUC3RVpqDSwBuw/NTweGyuP+o8AG98HxqxTBwID
+               AQABMA0GCSqGSIb3DQEBBQUAA4GBABS2TLuBeTPmcaTaUW/LCB2NYOy8GMdzR1mx
+               8iBIu2H6/E2tiY3RIevV2OW61qY2/XRQg7YPxx3ffeUugX9F4J/iPnnu1zAxxyBy
+               2VguKv4SWjRFoRkIfIlHX0qVviMhSlNy2ioFLy7JcPZb+v3ftDGywUqcBiVDoea0
+               Hn+GmxZA\n-----END CERTIFICATE-----""",
+        }
+    ]
+
+    test_return = {
+        "name": name,
+        "changes": {},
+        "result": None,
+        "comment": "Alias stringhost would have been updated\n",
+    }
+    state_return = {
+        "name": name,
+        "changes": {"stringhost": "Updated"},
+        "result": True,
+        "comment": "Alias stringhost updated.\n",
+    }
+    with patch.dict(keystore.__opts__, {"test": True}):
+        with patch.dict(
+            keystore.__salt__,
+            {
+                "keystore.list": MagicMock(return_value=cert_return),
+                "x509.read_certificate": MagicMock(return_value=x509_return),
+            },
+        ):
+            assert keystore.managed(name, passphrase, entries) == test_return
+
+    with patch.dict(
+        keystore.__salt__,
+        {
+            "keystore.list": MagicMock(return_value=cert_return),
+            "x509.read_certificate": MagicMock(return_value=x509_return),
+            "keystore.remove": MagicMock(return_value=True),
+            "keystore.add": MagicMock(return_value=True),
+        },
+    ):
+        assert keystore.managed(name, passphrase, entries) == state_return
+
+
+@patch("os.path.exists", MagicMock(return_value=False))
```

Comment:
Let's not use the `@patch` decorator please. Use `with patch()` inside the test function or use a fixture to do it outside of the test function.
https://engblog.nextdoor.com/what-your-mocks-do-when-you-aren-t-looking-b278e0d9e201

Reply:
The same for all other patch decorators in this module